includes/class-content-distribution.php

```diff
@@ -338,6 +339,44 @@ protected static function is_post_unlinked( $post_id ) {
 		return get_post_meta( $post_id, self::POST_UNLINKED_META, true );
 	}
 
+	/**
+	 * Upload the thumbnail for a linked post.
+	 *
+	 * @param int    $post_id       The linked post ID.
+	 * @param string $thumbnail_url The thumbnail URL.
+	 */
+	protected static function upload_thumbnail( $post_id, $thumbnail_url ) {
+		$post_payload         = get_post_meta( $post_id, self::POST_PAYLOAD_META, true );
+		$current_thumbnail_id = get_post_thumbnail_id( $post_id );
+
+		// Bail if the post has a thumbnail and the thumbnail URL is the same.
+		if (
+			$current_thumbnail_id &&
+			$post_payload &&
+			$post_payload['post_data']['thumbnail_url'] === $thumbnail_url
+		) {
+			return;
+		}
+
+		// Delete existing thumbnail attachment.
+		if ( $current_thumbnail_id ) {
+			wp_delete_attachment( $current_thumbnail_id, true );
+		}
```

Comment:
When syncing the post thumbnail, whether the distributed post is updating the featured image or deleting it, the destination site will permanently delete the orphan image.

I'm not sure if this is too aggressive, but I also don't like the idea of lingering orphan images.

The current approach can be problematic if:

- The site starts using the synced attachment in other content
- The site unlinked the post, changed the featured image to something local, and re-linked the post - deleting their local attachment

We can prevent the last one above by adding a meta to the synced attachment so only those get deleted.

Reply:
It's probably a pretty expensive query, but do you think it's worth checking if other posts are using the image before deleting it?

There is also the use case that the image is used in the post body.

Reply:
I've decided to not delete the attachment for now. Instead, we set a post meta to not lose track of the attachments that came via distribution and we can decide later how to deal with them. WDYT?

I think a post content query is not viable, even if executed via cron. It's very expensive...

Reply:
I think that makes sense. I like the idea with the postmeta. Better have a bit too much media than media missing.